server/src/main/java/org/candlepin/guice/CandlepinContextListener.java

```diff
@@ -259,6 +260,8 @@ public void contextDestroyed(ServletContextEvent event) {
     }
 
     private void destroySubsystems() throws Exception {
+        this.waitForArtemisJobThreadsToFinish();
```

Comment:
It was deliberately added before the JobManager.shutdown, because that method calls into these:
```
                this.dispatcher.shutdown();
                this.receiver.shutdown();
```
and the `receiver.shutdown()` is actually closing the sessions that these threads use. 
We don't want to close sessions of threads that are still running, but first wait for the threads to finish